### PR TITLE
Uniswap routes for swap tokens 

### PR DESCRIPTION
**Pull request by AjmalAhamedKodakkat**

### CODE REVIEW

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js [hunk 1]
```diff
@@ -0,0 +1,512 @@
...
+        const { address = "*" } = req.params;
+        const { privateKey = "", amount = "", tokenFrom = "", tokenTo = "" } = { ...req.query };
+        const tokenType = tokenTo.toString().toUpperCase();
+        const routerAbi = abi.UNISWAP_ROUTER.abi
+        const chainIds = ChainId.RINKEBY;
+
+        const tokenAddressRinkebyDai = '0xc7ad46e0b8a400bb3c915120d284aafba8fc4735'
+        const tokenAddressRinkebyUniswap = '0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984';
+        const tokenAddressRinkebyMaker = '0xF9bA5210F91D0474bd1e1DcDAeC4C58E359AaD85';
+        const tokenAddressRinkebyRouter = '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D'
```
Comment (srslafazan) [inline ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:35]: These will go in a map by network

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js [hunk 2]
```diff
@@ -0,0 +1,512 @@
...
+  ethereum: { web3, buildContract, compiler, abi },
+  ...context
+}) => {
+
+  router.post(
+    "/:address/swap/eth", [
+    param("address").trim().isString(),
+    query("tokenFrom").trim().isIn(["eth"]),
+    query("tokenTo").trim().isIn(["dai", "mkr", "uni"]),
+    query("privateKey").trim().isString(),
```
Comment (srslafazan) [inline ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:18]: We need network here

**Comment by srslafazan** [comment]
Looks wonderful

2 space indents ❤️ 

Need to switch on network for extensibility

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js [hunk 3]
```diff
@@ -0,0 +1,512 @@
...
+          case 'DAI':
+            token = DAI;
+            break;
+          case 'UNI':
+            token = UNISWAP;
+            break;
+          case 'MKR':
+            token = MAKER
+            break;
+        }
```
Comment (srslafazan) [inline ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:53]: We can probably use a map and a function for this

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js [hunk 4]
```diff
@@ -0,0 +1,512 @@
...
+  models,
+  ethereum: { web3, buildContract, compiler, abi },
+  ...context
+}) => {
+
+  router.post(
+    "/:address/swap/eth", [
+    param("address").trim().isString(),
+    query("tokenFrom").trim().isIn(["eth"]),
+    query("tokenTo").trim().isIn(["dai", "mkr", "uni"]),
```
Comment (srslafazan) [inline ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:17]: We should force these to either uppercase or lowercase (standard is typically uppercase for these)

**Review by srslafazan — APPROVED** [review]
LGTM -- The solution seems to be working as-is, and we may update it in a subsequent pull request. Another pull request may also extend mainnet support for the remaining methods.

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js [hunk 5]
```diff
@@ -0,0 +1,512 @@
...
+        const amountOutMin = trade.minimumAmountOut(slippageTolerance).raw
+        const path = [token.address, weth.address]
+        const deadline = Math.floor(Date.now() / 1000) + 60 * 20
+        const valueInput = trade.inputAmount.raw
+        const amountOutMinHex = ethers.BigNumber.from(amountOutMin.toString()).toHexString();
+        const value = ethers.BigNumber.from(valueInput.toString()).toHexString();
+        const wallet = web3.eth.accounts.wallet.add({
+          privateKey,
+          address,
+        });
```
Comment (srslafazan) [inline ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:149]: Many of these should be set as default values for user-provided parameters.

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js [hunk 6]
```diff
@@ -0,0 +1,512 @@
...
+          tokenAddressRouter = '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D'
+          tokenAddressWrappedEther = '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'
+        }
+        else if (networkType === "RINKEBY") {
+          tokenAddressDai = '0xc7ad46e0b8a400bb3c915120d284aafba8fc4735'
+          tokenAddressUniswap = '0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984';
+          tokenAddressMaker = '0xF9bA5210F91D0474bd1e1DcDAeC4C58E359AaD85';
+          tokenAddressRouter = '0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D'
+          tokenAddressWrappedEther = '0xc778417E063141139Fce010982780140Aa0cD5Ab';
+        }
```
Comment (srslafazan) [inline ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:333]: These constants can be lifted out of the logic circuit and into file or constant namespace

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js (hunk 1 above)
Comment (AjmalAhamedKodakkat) [reply ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:35]: this address will be taken as inputs in the coming version

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js (hunk 2 above)
Comment (AjmalAhamedKodakkat) [reply ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:18]: added network and chain validation

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js (hunk 3 above)
Comment (AjmalAhamedKodakkat) [reply ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:53]: this switch statements will be removed as we taken the token address as inputs

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js (hunk 4 above)
Comment (AjmalAhamedKodakkat) [reply ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:17]: done

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js (hunk 5 above)
Comment (AjmalAhamedKodakkat) [reply ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:149]: will consider in the future implementations

File: ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js (hunk 6 above)
Comment (AjmalAhamedKodakkat) [reply ethereum-rpc/src/constructors/express/routes/api/v1/uniswap.js:333]: these constants will be invalid in the coming version , will try to set constants in the share folder in future implementations

**Pull request merged**